core/src/main/java/org/stellar/anchor/sep38/Sep38Service.java

```diff
@@ -194,16 +195,30 @@ public GetPriceResponse getPrice(
     GetPriceResponse.GetPriceResponseBuilder builder =
         GetPriceResponse.builder().price(rateResponse.getRate().getPrice());
     BigDecimal bPrice = new BigDecimal(rateResponse.getRate().getPrice());
+    BigDecimal bSellAmount, bBuyAmount;
     if (sellAmount != null) {
-      BigDecimal bSellAmount = new BigDecimal(sellAmount);
-      BigDecimal bBuyAmount = bSellAmount.divide(bPrice, 4, RoundingMode.HALF_UP);
-      builder = builder.sellAmount(sellAmount).buyAmount(bBuyAmount.toPlainString());
+      bSellAmount = new BigDecimal(sellAmount);
+      bBuyAmount = bSellAmount.divide(bPrice, buyAsset.getDecimals(), RoundingMode.HALF_UP);
```

Comment:
❓ Shouldn't we round down? If the anchor has specified 2 decimals of precision, and we're dealing with an amount 1.234, shouldn't the service simply not recognize the 3rd decimal and act as if 1.23 was sent?

There are a few other places `HALF_UP` is used.

Reply:
Yeah, great point!

Added https://github.com/stellar/java-stellar-anchor-sdk/pull/125/commits/eb69c3657b34be78a0e8f201cbb5deab0504466d to update the rounding logic:
* sell_amount is rounded UP
* buy_amount is rounded DOWN

let me know if that makes sense to you too.

Reply:
Why is `sell_amount` rounded up?

Reply:
If we consider 2 parties, CLIENT and ANCHOR, it's expected that the volume of ANCHOR transactions is way higher than CLIENT's. With that in mind, I propose the rounding difference be shifted in favor of the ANCHOR to prevent the risk where a high amount of transactions could cause a significant loss in the long term.

Let's take the following scenarios (considering 2 decimals only)

Rounding sell amount UP will prevent the ANCHOR from receiving less than expected:
* `sell_amount=1.354` --rounded to--> `sell_amount=1.36` (UP)
* `buy_amount=2.400` --rounded to-->`buy_amount=2.40`

Rounding buy amount DOWN will prevent the ANCHOR from paying more than expected:
* `sell_amount=3.620` --rounded to--> `sell_amount=3.62`
* `buy_amount=6.407` --rounded to-->`buy_amount=6.40` (DOWN)

WDYT?

Reply:
I'm a bit confused with the examples, but I think I understand what you're saying.

If a client requests a `sell_amount` of `1.2345` (say 4 decimals are allowed for this off-chain asset), and the `price` (exchange rate) is `2.3456`, then `buy_amount` is `0.52630457...`.

Lets say the buy asset uses 5 decimals (even though a Stellar asset would use 7). If we round UP, `buy_amount` is `0.52631.`. If we round DOWN, `buy_amount` is `0.52630`.

In this case, it would be better for the anchor if we rounded DOWN like you suggest. In the same way, when calculating `sell_amount` it would be better for the anchor if we rounded UP.

I think this makes sense. Even though the platform is being opinionated by choosing a rounding strategy, the anchor ultimately has the ability to adjust their `price` if they would rather give the client a favorable rate.

Reply:
I updated all rounding to HALF_DOWN to follow common practices.